File: 动态规划专题/1137_第 N 个泰波那契数.py

#动态规划
class Solution:
    def tribonacci(self, n: int) -> int:
        f = []
        f.append(0)
        f.append(1)
        f.append(1)
        if n > 2:
            for i in range(3, n + 1):
                f.append(f[i - 3] + f[i - 2] + f[i - 1])
        return f[n]

File: 动态规划专题/1137_第 N 个泰波那契数.c
//1137_第 N 个泰波那契数

//动态规划
int tribonacci(int n) {
	int i = 0;
	int f[38] = { 0 };
	f[0] = 0;
	f[1] = 1;
	f[2] = 1;
	if (n > 2)
	{
		for (i = 3; i < n + 1; i++)
		{
			f[i] = f[i - 3] + f[i - 2] + f[i - 1];
		}
	}
	return f[n];
}